compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -1819,8 +1925,63 @@ private void analyzeInvocation(BLangInvocation invocationExpr) {
         taintedStatus = returnTaintedStatus;
     }
 
+    private void updateArgTaintedStatus (BLangInvocation invocationExpr, List<TaintedStatus> argTaintedStatusList) {
+        if (!argTaintedStatusList.isEmpty()) {
+            BInvokableSymbol invokableSymbol = (BInvokableSymbol) invocationExpr.symbol;
+            int requiredParamCount = invokableSymbol.params.size();
+            int defaultableParamCount = invokableSymbol.defaultableParams.size();
+
+            for (int argIndex = 0; argIndex < invocationExpr.requiredArgs.size(); argIndex++) {
+                BLangExpression argExpr = invocationExpr.requiredArgs.get(argIndex);
+                TaintedStatus argTaintedStatus = TaintedStatus.UNTAINTED;
+                if (!argTaintedStatusList.isEmpty()) {
+                    argTaintedStatus = argTaintedStatusList.get(argIndex);
+                }
+                updateArgTaintedStatus(argExpr, argTaintedStatus);
+            }
+
+            for (int argIndex = 0; argIndex < invocationExpr.namedArgs.size(); argIndex++) {
+                BLangExpression argExpr = invocationExpr.namedArgs.get(argIndex);
+                if (argExpr.getKind() == NodeKind.NAMED_ARGS_EXPR) {
+                    String currentNamedArgExprName = ((BLangNamedArgsExpression) argExpr).name.value;
+                    // Pick the index of this defaultable parameter in the invokable definition.
+                    int paramIndex = 0;
+                    for (int defaultableParamIndex = 0; defaultableParamIndex < invokableSymbol.defaultableParams
```

Comment:
Rather than getting invokableSymbol.defaultableParams.size() in iteration of loop we can use the local variable defaultableParamCount here. Consider the same for other for loops as well.